src/mediapipe_internal/mediapipegraphexecutor.cpp

```diff
@@ -188,32 +212,87 @@ Status MediapipeGraphExecutor::infer(const KFSRequest* request, KFSResponse* res
                 name, request->model_name(), absMessage);
             return Status(StatusCode::MEDIAPIPE_GRAPH_CLOSE_INPUT_STREAM_ERROR, std::move(absMessage));
         }
-    }
-    // receive outputs
-    ::mediapipe::Packet packet;
-    std::set<std::string> outputPollersWithReceivedPacket;
-    for (auto& [outputStreamName, poller] : outputPollers) {
-        size_t receivedOutputs = 0;
-        SPDLOG_DEBUG("Will wait for output stream: {} packet", outputStreamName);
-        while (poller.Next(&packet)) {
-            SPDLOG_DEBUG("Received packet from output stream: {}", outputStreamName);
-            auto received = packet.Get<ov::Tensor>();
-            auto* output = response->add_outputs();
-            output->set_name(outputStreamName);
-            output->set_datatype(
-                ovmsPrecisionToKFSPrecision(
-                    ovElementTypeToOvmsPrecision(
-                        received.get_element_type())));
-            output->clear_shape();
-            for (const auto& dim : received.get_shape()) {
-                output->add_shape(dim);
+
+        // Size checked to be equal 1 at the beggining of the function
+        for (auto& [outputStreamName, poller] : outputPollers) {
+            size_t receivedOutputs = 0;
+            SPDLOG_DEBUG("Will wait for output stream: {} packet", outputStreamName);
+            while (poller.Next(&packet)) {
+                SPDLOG_DEBUG("Received packet from output stream: {}", outputStreamName);
+                try {
+                    auto received = packet.Get<KFSResponse*>();
+                    if (received == nullptr) {
+                        SPDLOG_DEBUG("Received nullptr KFSResponse for: {}", outputStreamName);
+                        continue;
+                    }
+
+                    *response = std::move(*received);
+                } catch (const std::exception& e) {
+                    SPDLOG_DEBUG("Mediapipe 'packet.Get' exception {}", e.what());
```

Comment:
When this exception happens?

Reply:
Null ptr dereference AFAIK.

Reply:
This could happen if calculator is poorly written?

Reply:
Yes.